src/scrollNav.js

```diff
@@ -97,7 +98,16 @@
         var section_id    = 'scrollNav-' + (i + 1);
         var isFirst       = function() { return i === 0; };
         var hasHeading    = function() { return !$this_section.eq(0).is(S.settings.sections); };
-        var text          = ( S.settings.showTopLink && isFirst() && hasHeading() ) ? S.settings.topLinkText : $this_section.filter(S.settings.sections).text();
+
+        var text;
+
+        if ( S.settings.showTopLink && isFirst() && hasHeading() ) {
+          text = S.settings.topLinkText;
+        } else if ( $this_section.filter(S.settings.sections).filter('[data-' + S.settings.labelAttribute + ']').length && $this_section.filter(S.settings.sections).data(S.settings.labelAttribute) !== "" ) {
```

Comment:
Could this condition be simplified? It's doing a lot of dom querying. Also, your instructions say the `data` prefix is optional but you've assumed it's used here. Let's make the value in the condition the exact value passed through the settings, so that the prefix is truly optional. Lastly, please stick to single quotes.